src/Reflection/ReflectionParameter.php

```diff
@@ -541,11 +541,17 @@ public function getClass() : ?ReflectionClass
             return null;
         }
 
+        $fqsen = $hint->getFqsen();
+
+        if (! $fqsen) {
```

Comment:
the cyclomatic complexity:

![too-damn-high](https://user-images.githubusercontent.com/154256/28508937-b5bf0b1a-703e-11e7-861b-77ed07c15cc2.jpg)